### PR TITLE
Tests/validators

### PR DESCRIPTION
**Pull request by xRoms**

### CODE REVIEW

**Review by AlexSaplin — CHANGES_REQUESTED** [review]

File: test/CowswapValidator.test.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                                randomAddress(),
+                                generateSingleParams(uint256),
+                                PRE_SIGNATURE_SELECTOR,
+                                randomBytes(32)
+                            )
+                    ).to.not.be.reverted;
+                });
+            });
+            describe("edge cases:", async () => {
+                describe("if selector is not pre_signature", async () => {
```
[the remark this answers is not in the document]
Comment (xRoms) [reply test/CowswapValidator.test.ts:62]: ✓

File: test/behaviors/validator.ts [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        it("true for IValidator", async () => {
+            await withSigner(this.admin.address, async (signer) => {
+                await expect(
+                    await this.subject
+                        .connect(signer)
+                        .supportsInterface(VALIDATOR_INTERFACE_ID)
+                ).to.be.true;
+            });
+        });
+        it("false for IVault", async () => {
```
[the remark this answers is not in the document]
Comment (xRoms) [reply test/behaviors/validator.ts:257]: ✓

File: test/behaviors/validator.ts [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+                    await expect(
+                        this.subject.connect(signer).commitValidatorParams()
+                    ).to.not.be.reverted;
+                });
+            });
+        });
+    });
+
+    describe("#supportsInterface", () => {
+        it("true for IValidator", async () => {
```
[the remark this answers is not in the document]
Comment (xRoms) [reply test/behaviors/validator.ts:248]: ✓

**Review by AlexSaplin — APPROVED** [review]
